src/main/Java/jonathan/Jonathan.java

```diff
@@ -0,0 +1,257 @@
+package jonathan;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Objects;
+import java.util.Scanner;
+import java.io.File;
+import java.io.FileWriter;
+
+public class Jonathan {
+    public static void main(String[] args) {
+
+
+        Scanner input = new Scanner((System.in));
+        String command;
+        String line = "    ____________________________________________________________";
+        String welcome = line + "\n    Hello! I'm Jonathan\n" +
+                "    What can I do for you?\n" + line + "\n";
+        String goodbye = line + "\n    Bye. Hope to see you again soon!\n" + line;
+
+        ArrayList<Task> list = readData();
+        int taskCounter = list.size();
+
+        System.out.println(welcome);
+
+        // Loop until user prompt "bye"
+        while (true) {
+            command = input.nextLine(); // Prompt input from user
+
+            try {
+                if (Objects.equals(command, "bye")) {
+                    saveData(list);
+                    System.out.println(goodbye);
+                    break; // quit the program
+
+                } else if (Objects.equals(command, "list")) {
+                    lookupTask(list, taskCounter);
+
+                } else if (command.split(" ")[0].equals("mark") || command.split(" ")[0].equals("unmark")) {
```

Comment:
This is a fairly complex statement if code readability is taken to account. To improve this readability, you should consider simplifying it by referring to the coding readability standards. Link - https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-complicated-expressions